src/cosalib/cmdlib.py

```diff
@@ -72,13 +74,23 @@ def write_json(path, data):
     :type: path: str
     :param data:  structure to write out as json
     :type data: dict or list
+    :param lock_file: Exclusively lock file before writing
+    :type: lock_file: bool
     :raises: ValueError, OSError
     """
     dn = os.path.dirname(path)
     f = tempfile.NamedTemporaryFile(mode='w', dir=dn, delete=False)
     json.dump(data, f, indent=4)
     os.fchmod(f.file.fileno(), 0o644)
-    shutil.move(f.name, path)
+
+    # lock before moving
+    bn = os.path.basename(path)
+    lock = Lock(f"{dn}/.{bn}.lock")
+    try:
+        with lock:
+            shutil.move(f.name, path)
+    finally:
+        lock.unlock()
```

Comment:
```suggestion
    with Lock(f"{dn}/.{bn}.lock"):
        shutil.move(f.name, path)
```

Reply:
@bgilbert I caught this the other day 😄 https://github.com/darkmuggle/coreos-assembler/blob/e8a46ac5a7b8019a17745bb17da7a362ec7e0668/src/cosalib/cmdlib.py#L89-L90